FogCreek/Glitch-Community
# Heady path: use personalized search credentials to remove private projects

## Links
* Remix link: https://heady-path.glitch.me

## Changes:
* use `/search/creds` to get a personalized algolia API key
* rearrange `useAlgoliaSearch` to incorporate the `/search/creds` request 
* correctly render private projects in search results

## How To Test:
* while logged out, enable algolia search at  https://heady-path.glitch.me/secret
* search for `glitchdb` (a private glitch project) https://heady-path.glitch.me/search?q=glitchdb
* it should not be visible
* log in (by setting `localStorage.cachedUser` to include your persistent token)
* search for `glitchdb` again https://heady-path.glitch.me/search?q=glitchdb
* it should now be visible

## Feedback I'm looking for:
* is this correctly showing/hiding private projects?
